src/analysis/normed_space/exponential.lean

```diff
@@ -0,0 +1,551 @@
+lemma has_fderiv_at_exp_zero_of_radius_pos (h : 0 < (exp_series 𝕂 𝔸).radius) :
+  has_fderiv_at (exp 𝕂 𝔸) (1 : 𝔸 →L[𝕂] 𝔸) 0 :=
+(has_strict_fderiv_at_exp_zero_of_radius_pos h).has_fderiv_at
+
+/-- In a Banach-algebra `𝔸` over a normed field `𝕂` of characteristic zero, if `x` and `y` are
+in the disk of convergence and commute, then `exp 𝕂 𝔸 (x+y) = (exp 𝕂 𝔸 x) * (exp 𝕂 𝔸 y)`. -/
+lemma exp_add_of_commute_of_lt_radius [char_zero 𝕂]
+  {x y : 𝔸} (hxy : commute x y)
+  (hx : ↑∥x∥₊ < (exp_series 𝕂 𝔸).radius) (hy : ↑∥y∥₊ < (exp_series 𝕂 𝔸).radius) :
+  exp 𝕂 𝔸 (x + y) = (exp 𝕂 𝔸 x) * (exp 𝕂 𝔸 y) :=
+begin
+  have : ∀ {t : 𝔸}, ↑∥t∥₊ < (exp_series 𝕂 𝔸).radius → summable (λ n : ℕ, ∥(1 / n! : 𝕂) • t ^ n∥),
+  { intros t ht,
+    conv {congr, funext, rw ← exp_series_apply_eq},
+    exact (exp_series 𝕂 𝔸).summable_norm_of_nnnorm_lt_radius ht },
+  rw [exp_eq_tsum, tsum_mul_tsum_eq_tsum_sum_antidiagonal_of_summable_norm (this hx) (this hy)],
+  dsimp only,
+  conv_lhs {congr, funext, rw [hxy.add_pow' _, finset.smul_sum]},
+  refine tsum_congr (λ n, finset.sum_congr rfl $ λ kl hkl, _),
+  have : kl.1 ≤ n := finset.nat.antidiagonal.fst_le hkl,
+  rw [nsmul_eq_smul_cast 𝕂, smul_smul, smul_mul_smul, @nat.cast_choose _ _ ‹_› _ _ this,
+      nat.sub_eq_of_eq_add (finset.nat.mem_antidiagonal.mp hkl).symm],
+  congr' 1,
+  have : (n! : 𝕂) ≠ 0 := nat.cast_ne_zero.mpr n.factorial_ne_zero,
+  field_simp [this]
+end
+
+end complete_algebra
+
+end any_field_any_algebra
+
+section any_field_comm_algebra
+
+variables {𝕂 𝔸 : Type*} [nondiscrete_normed_field 𝕂] [normed_comm_ring 𝔸] [normed_algebra 𝕂 𝔸]
+  [complete_space 𝔸]
+
+/-- In a commutative Banach-algebra `𝔸` over a normed field `𝕂` of characteristic zero,
+`exp 𝕂 𝔸 (x+y) = (exp 𝕂 𝔸 x) * (exp 𝕂 𝔸 y)` for all `x`, `y` in the disk of convergence. -/
+lemma exp_add_of_lt_radius [char_zero 𝕂] {x y : 𝔸}
+  (hx : ↑∥x∥₊ < (exp_series 𝕂 𝔸).radius) (hy : ↑∥y∥₊ < (exp_series 𝕂 𝔸).radius) :
+  exp 𝕂 𝔸 (x + y) = (exp 𝕂 𝔸 x) * (exp 𝕂 𝔸 y) :=
+exp_add_of_commute_of_lt_radius (commute.all x y) hx hy
+
+/-- The exponential map in a commutative Banach-algebra `𝔸` over a normed field `𝕂` of
+characteristic zero has Fréchet-derivative `exp 𝕂 𝔸 x • 1 : 𝔸 →L[𝕂] 𝔸` at any point `x` in the
+disk of convergence. -/
+lemma has_fderiv_at_exp_of_lt_radius [char_zero 𝕂] {x : 𝔸} (hx : ↑∥x∥₊ < (exp_series 𝕂 𝔸).radius) :
+  has_fderiv_at (exp 𝕂 𝔸) (exp 𝕂 𝔸 x • 1 : 𝔸 →L[𝕂] 𝔸) x :=
+begin
+  have hpos : 0 < (exp_series 𝕂 𝔸).radius := (ennreal.coe_nonneg.mpr $ zero_le _).trans_lt hx,
+  rw has_fderiv_at_iff_is_o_nhds_zero,
+  suffices : (λ ε, exp 𝕂 𝔸 x * (exp 𝕂 𝔸 (0 + ε) - exp 𝕂 𝔸 0 - continuous_linear_map.id 𝕂 𝔸 ε))
+    =ᶠ[𝓝 0] (λ ε, exp 𝕂 𝔸 (x + ε) - exp 𝕂 𝔸 x - exp 𝕂 𝔸 x • continuous_linear_map.id 𝕂 𝔸 ε),
+  { refine (is_o.const_mul_left _ _).congr' this (eventually_eq.refl _ _),
+    rw ← has_fderiv_at_iff_is_o_nhds_zero,
+    exact has_fderiv_at_exp_zero_of_radius_pos hpos },
+  have : ∀ᶠ ε : 𝔸 in 𝓝 0, ↑∥ε∥₊ < (exp_series 𝕂 𝔸).radius,
```

Comment:
```suggestion
  have : ∀ᶠ ε in 𝓝 (0 : 𝔸), ↑∥ε∥₊ < (exp_series 𝕂 𝔸).radius,
```
I find this slightly more legible (and I would also use `h` instead of `ε`).